src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+    {
+
+        #region Performance test configuration
+
+        public static IEnumerable<object[]> PerformanceMemberData(SerializerType serializerType)
+        {
+            yield return new object[] { 100, TestType.ByteArray, 1024, serializerType };
+            yield return new object[] { 5, TestType.ByteArray, 1024 * 1024, serializerType };
+            yield return new object[] { 10000, TestType.String, 128, serializerType };
+            yield return new object[] { 10000, TestType.String, 1024, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 128, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 1024, serializerType };
+            yield return new object[] { 1, TestType.ListOfInt, 1024 * 1024, serializerType };
+            yield return new object[] { 1000, TestType.Dictionary, 128, serializerType };
+            yield return new object[] { 100, TestType.Dictionary, 1024, serializerType };
+            yield return new object[] { 10, TestType.SimpleType, 1, serializerType };
+            yield return new object[] { 1, TestType.SimpleType, 15, serializerType };
+            yield return new object[] { 10000, TestType.ISerializable, -1, serializerType };
+            yield return new object[] { 10000, TestType.XmlElement, -1, serializerType };
+        }
+
+        #endregion
+
+        #region Methods to create object for serialization tests
+
+        public static SimpleTypeWihtMoreProperties CreateObject(int height, int parentId, int currentId, int collectionSize, int childListSize)
+        {
+            int index = parentId * childListSize + (currentId + 1);
+            var obj = new SimpleTypeWihtMoreProperties()
+            {
+                IntProperty = index,
+                StringProperty = index + " string value",
+                EnumProperty = (MyEnum)(index % (Enum.GetNames(typeof(MyEnum)).Length)),
+                CollectionProperty = new List<string>(),
+                SimpleTypeList = new List<SimpleTypeWihtMoreProperties>()
+            };
+            for (int i = 0; i < collectionSize; ++i)
+            {
+                obj.CollectionProperty.Add(index + "." + i);
+            }
+            if (height > 1)
+            {
+                for (int i = 0; i < childListSize; ++i)
+                {
+                    obj.SimpleTypeList.Add(CreateObject(height - 1, index, i, collectionSize, childListSize));
+                }
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>();
```

Comment:
Pass the count to the Dictionary constructor. It defaults to a size of 3, and needs to grow (which requires new allocation and copying) each time the dictionary gets close to capacity. This could skew your results by requiring extra GC work during the serialization/deserialization.